File: frontend/src/services/demandService.js
import { httpGet } from '@/api/client';
import { API } from '@/constants/apiEndpoints';

export const demandService = {
  async getForecasts() {
    const res = await httpGet(API.DEMAND.FORECAST);
    return Array.isArray(res) ? res : res.forecasts ?? [];
  },

  async getForecastForProduct(productId) {
    return httpGet(`${API.DEMAND.FORECAST}/${productId}`);
  },
};

File: backend/src/main/java/com/agrolink/service/DemandService.java
package com.agrolink.service;

import com.agrolink.dto.demand.DemandForecastResponse;
import com.agrolink.dto.demand.DemandListResponse;

public interface DemandService {

    DemandListResponse forecasts();

    DemandForecastResponse forecastFor(Long productId);
}